### PR TITLE
Add support to api packages in Application Broker and Application CRD

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

Add support to api packages in Application Broker:
- Support for multiple GATEWAY and TARGET url
    - Pattern: `upperCase(<APP_NAME>)_GATEWAY_URL`, `upperCase(<APP_NAME>)_TARGET_URL`
- Remove restriction “provision only once”
- Implemented on feature flag `APP_API_PACKAGES_SUPPORT`
- Remove the OwnerReference from the EventActivation and replace that with removing that on deprovision action. (feature changed in both approaches with api-packages and without them)
- Fix updating the Istio Policy on provision action

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/7201

### CODE REVIEW

**Comment by mszostok** [comment]
here is a task in which the old impl should be removed: https://github.com/kyma-project/kyma/issues/7415

the `TODO` statement in source code will be updated after review

File: components/application-broker/internal/syncer/validator.go [hunk 1]
```diff
@@ -8,11 +8,74 @@ import (
...
+				}
+
+			case eventEntryType:
+				EventEntryCnt++
+			}
+		}
+
+		if APIEntryCnt == 0 && EventEntryCnt == 0 {
+			messages = append(messages, fmt.Sprintf("Service with id %q is invalid. Requires at least one API or Event entry", svc.ID))
+		}
```
Comment (jasiu001) [inline components/application-broker/internal/syncer/validator.go]: First, number of Entries is check (`len(svc.Entries) == 0`) then the concrete amount of type is checked, it means we can expect other not supported types so maybe it is good to log (`default case`) other types which can appear in Entries

File: components/application-broker/internal/syncer/validator.go [hunk 2]
```diff
@@ -8,11 +8,74 @@ import (
...
+			return true
+		}
+	}
+	return false
+}
+
+// Deprecated
 type appCRValidator struct{}
 
 const (
```
Comment (jasiu001) [inline components/application-broker/internal/syncer/validator.go]: move const up

File: components/application-broker/internal/broker/error.go [hunk 3]
```diff
@@ -1,8 +1,17 @@
 package broker
 
+import "github.com/pkg/errors"
+
+type notFoundError struct{}
```
Comment (jasiu001) [inline components/application-broker/internal/broker/error.go]: Why this error is repated? Check `/internal/storage/driver/memory/error.go`, IMO it should be in one place.

File: components/application-broker/internal/model.go [hunk 4]
```diff
@@ -10,44 +11,83 @@ type ApplicationName string
...
 	Labels map[string]string
+}
 
-	//TODO(entry-simplification): this is an accepted simplification until
-	// explicit support of many APIEntry and EventEntry
-	APIEntry      *APIEntry
-	EventProvider bool
+func (s *Service) IsBindable() bool {
+	for _, e := range s.Entries {
+		if e.Type == "API" {
```
Comment (jasiu001) [inline components/application-broker/internal/model.go]: Why don't you compare it with some const, e.g. `apiEntryType` from validator file? 
By the way, the same values (API, Events) are in both places `mapper.go` and `validator.go`.

Comment (mszostok) [reply components/application-broker/internal/model.go]: I've added proper const in `model.go` file and used them across all packages

File: components/application-broker/internal/broker/error.go (hunk 3 above)
Comment (mszostok) [reply components/application-broker/internal/broker/error.go]: I forget to remove it :)

**Pull request merged**